src/ucs/type/status.h

```diff
@@ -113,6 +113,12 @@ typedef void *ucs_status_ptr_t;
 #define UCS_STATUS_IS_ERR(_status) ((_status) < 0)
 
 
+#define UCS_IF_STATUS_IS_NOT_OK(_status, _action) \
```

Comment:
IMO this is an overkill..

Reply:
just noticed that we have a ton of such code like
```
if (status != UCS_OK) {
    ucs_error(...); // optional
    return status;
}
```
it's not related to this PR directly but could to reduce many lines of new code, refactorind of legacy looks overkill to me too

Reply:
the macro is reducing few lines of code but I think it's less readable this way